src/agent/src/rpc.rs

```diff
@@ -2101,18 +2101,40 @@ async fn cdh_handler(oci: &mut Spec) -> Result<()> {
         .ok_or_else(|| anyhow!("Spec didn't contain mounts field"))?;
 
     for m in mounts.iter_mut() {
-        if m.destination().starts_with("/sealed") {
-            info!(
-                sl(),
-                "sealed mount destination: {:?} source: {:?}",
-                m.destination(),
-                m.source()
-            );
-            if let Some(source_str) = m.source().as_ref().and_then(|p| p.to_str()) {
-                cdh::unseal_file(source_str).await?;
-            } else {
-                warn!(sl(), "Failed to unseal: Mount source is None or invalid");
+        if let Some(source_path) = m.source().as_ref().and_then(|p| p.to_str()) {
+            // Check if source_path starts with "/run/kata-containers/shared/containers"
+            // For a volume mount path /mydir,
+            // the secret file path will be like this under the /run/kata-containers/shared/containers dir
+            // a128482812bad768f404e063f225decd425fc94a673aec4add45a9caa1122ccb-75490e32e51da3ff-mydir
+            // We can ignore few paths like: resolv.conf, termination-log, hostname,hosts,serviceaccount
+            if source_path.starts_with("/run/kata-containers/shared/containers")
```

Comment:
As my understanding, we should check the mount whether it is a sealed_secret volume, right ?
What's about introducing a dedicated method which wrapping the checking logic ?

Reply:
@Apokleos thanks for your comments. Yes, the check can be made part of a dedicated method which can be improved iteratively.

Reply:
@Apokleos Added helper function for the path check. PTAL